projects/spectator/src/lib/matchers.ts

```diff
@@ -340,18 +340,29 @@ function isHidden(elOrSelector: HTMLElement | string): boolean {
     return true;
   }
 
+  const hiddenWhen = [
+    el => !(el.offsetWidth || el.offsetHeight || el.getClientRects().length),
+    el => el.style.display === 'none',
+    el => el.style.visibility === 'hidden',
+    el => el.type === 'hidden',
+    el => el.hasAttribute('hidden')
+  ];
+
+  const runningInJsDom = navigator.userAgent.includes('Node.js') || navigator.userAgent.includes('jsdom');
```

Comment:
Let's move it to an export function so that we re-use it in the future with other special cases.

Reply:
Agreed, this is most likely not the last time that JSDOM differs from the a browser DOM.